### PR TITLE
RequestType .urlEncoded use percent encoding

### PR DESCRIPTION
**Pull request by eRDe33**

### CODE REVIEW

File: Sources/URLRequest+APIAdapter.swift [hunk 1]
```diff
@@ -53,9 +53,19 @@ extension URLRequest {
     }
 
     private mutating func setURLEncoded(parameters: HTTPParameters) {
+        let allowedCharacters = CharacterSet(charactersIn: "!*'();:@&=+$,/?%#[] ").inverted
+        var queryItems = [URLQueryItem]()
+        parameters.forEach { (key, value) in
```
Comment (mkj-is) [inline Sources/URLRequest+APIAdapter.swift]: In APIKit we mostly use functional methods for transforming collections. I propose using `compactMap`.

```suggestion
        let queryItems: [URLQueryItem] = parameters.compactMap { (key, value) in
```

File: Sources/URLRequest+APIAdapter.swift [hunk 2]
```diff
@@ -53,9 +53,19 @@ extension URLRequest {
...
+            guard let encodedKey = key.addingPercentEncoding(withAllowedCharacters: allowedCharacters),
+                let encodedValue = value.addingPercentEncoding(withAllowedCharacters: allowedCharacters) else {
+                    return
+            }
+            queryItems.append(URLQueryItem(name: encodedKey, value: encodedValue))
```
Comment (mkj-is) [inline Sources/URLRequest+APIAdapter.swift]:
```suggestion
            return URLQueryItem(name: encodedKey, value: encodedValue)
```

File: Sources/URLRequest+APIAdapter.swift [hunk 3]
```diff
@@ -53,9 +53,19 @@ extension URLRequest {
     }
 
     private mutating func setURLEncoded(parameters: HTTPParameters) {
+        let allowedCharacters = CharacterSet(charactersIn: "!*'();:@&=+$,/?%#[] ").inverted
+        var queryItems = [URLQueryItem]()
+        parameters.forEach { (key, value) in
+            guard let encodedKey = key.addingPercentEncoding(withAllowedCharacters: allowedCharacters),
+                let encodedValue = value.addingPercentEncoding(withAllowedCharacters: allowedCharacters) else {
+                    return
```
Comment (mkj-is) [inline Sources/URLRequest+APIAdapter.swift]:
```suggestion
                    return nil
```

File: Sources/URLRequest+APIAdapter.swift [hunk 4]
```diff
@@ -53,9 +53,19 @@ extension URLRequest {
     }
 
     private mutating func setURLEncoded(parameters: HTTPParameters) {
+        let allowedCharacters = CharacterSet(charactersIn: "!*'();:@&=+$,/?%#[] ").inverted
```
Comment (mkj-is) [inline Sources/URLRequest+APIAdapter.swift]: Shouldn't be all non-ASCII characters encoded too?

```suggestion
        let allowedCharacters = CharacterSet(charactersIn: "!*'();:@&=+$,/?%#[]").inverted
```

File: Sources/URLRequest+APIAdapter.swift [hunk 5]
```diff
@@ -53,9 +53,19 @@ extension URLRequest {
...
+        var queryItems = [URLQueryItem]()
```
Comment (mkj-is) [inline Sources/URLRequest+APIAdapter.swift]:
```suggestion
```

File: Sources/URLRequest+APIAdapter.swift (hunk 4 above)
Comment (eRDe33) [reply Sources/URLRequest+APIAdapter.swift]: I don't know why, but non-ASCII characters (like £, ¢, ∞, §, etc.) are encoded too. 😮 ...And to your suggestion, I think the white-space should be encoded too. As not allowed characters I choosed URI reserved characters (by RFC 3986) and white-space.

**Pull request merged**